packages/app/src/messages/components/messages/messages.component.tsx

```diff
@@ -604,14 +599,41 @@ const Messages: FunctionComponent<MessagesProps> = ({
     allItemsSelected ? resetItems() : selectAllItems()
   }
 
-  const handleSearchSelect = (thread: Thread) => {
-    setActiveThread(thread)
-    setMessagesState(MessagesState.ThreadDetails)
+  const handleSearchSelect = (record: Thread | Message) => {
+    if (!record) {
+      return
+    }
+
+    if ((record as Message).threadId) {
```

Comment:
Small thing, but type guard would be good idea here. Something like:
```
    const isMessage = (record: Thread | Message): record is Message => {
      return Boolean((record as Message).threadId)
    }
```

Reply:
https://github.com/mudita/mudita-center/pull/945/commits/735a58b6787a26a9f9fc3714fa5138d4dc264f41